api/rest/coordinator.go

```diff
@@ -158,6 +166,18 @@ func (c ParallelCoordinator) CollectBundle(ctx context.Context, bundleID string,
 		bundlePaths = append(bundlePaths, bundlePath)
 	}
 
+	// Run cleanup in separated goroutine so it will not block bundle generation process
+	go func() {
+		for _, b := range bundlesToDelete {
```

Comment:
Can we be sure that `bundlesToDelete == numBundles` at this point? Otherwise you'd have to check whether `b` is the zero value here.

Reply:
Yes we can, there is a simple for loop https://github.com/dcos/dcos-diagnostics/pull/147/files/2964a8cf0fd0675cfb6b69c3ed0747da7423b5a9#diff-7b54bfd9ba87b3e1cbcc388b026985d7R138 with no `break`